cmd/object-handlers.go

```diff
@@ -2152,25 +2152,42 @@ func (api objectAPIHandlers) ListObjectPartsHandler(w http.ResponseWriter, r *ht
 	writeSuccessResponseXML(w, encodedSuccessResponse)
 }
 
-// Send white spaces every 10 seconds to the client till completeMultiPartUpload() is done so that the client does not time out.
-// Downside is we might send 200OK and then send error XML. But accoording to S3 spec
-// the client is supposed to check for error XML even if it received 200OK. But for erasure this is not a
-// problem as completeMultiPartUpload() is quick. Even For FS, it would not be an issue
-// as we do background append as and when the parts arrive and completeMultiPartUpload is quick.
-// Only in a rare case where parts would be out of order will FS' completeMultiPartUpload() take a longer time.
-func sendWhiteSpace(ctx context.Context, w http.ResponseWriter) <-chan struct{} {
+type emptyCharWriter struct {
+	http.ResponseWriter
+	http.Flusher
+	written bool
+}
+
+func (e *emptyCharWriter) Write(b []byte) (int, error) {
+	n, err := e.ResponseWriter.Write(b)
+	e.written = true
+	return n, err
+}
+
+func (e *emptyCharWriter) WriteHeader(statusCode int) {
+	if !e.written {
+		e.ResponseWriter.WriteHeader(statusCode)
+	}
+}
+
+// Send empty chars every 10 seconds to the client till completeMultiPartUpload() is
+// done so that the client does not time out. Downside is we might send 200 OK and
+// then send error XML. But accoording to S3 spec the client is supposed to check
+// for error XML even if it received 200 OK. But for erasure this is not a problem
+// as completeMultiPartUpload() is quick. Even For FS, it would not be an issue as
+// we do background append as and when the parts arrive and completeMultiPartUpload
+// is quick. Only in a rare case where parts would be out of order will
+// FS:completeMultiPartUpload() take a longer time.
+func sendEmptyChar(ctx context.Context, w http.ResponseWriter) <-chan struct{} {
 	doneCh := make(chan struct{})
 	go func() {
 		ticker := time.NewTicker(time.Second * 10)
 		for {
 			select {
 			case <-ticker.C:
-				// Write a white space char to the client to prevent client timeouts
+				// Write empty char to the client to prevent client timeouts
 				// when the server takes a long time to completeMultiPartUpload()
-				if _, err := w.Write([]byte("\n\r")); err != nil {
-					logger.LogIf(ctx, err)
-					return
-				}
+				w.Write([]byte(""))
```

Comment:
there will be no side-effect because of this, i.e there wont be any data transferred on the network because of this call and hence client will still timeout.

Reply:
Fixed